src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicBuffers.cs

```diff
@@ -37,44 +33,35 @@ private void FreeNativeMemory()
         QUIC_BUFFER* buffers = _buffers;
         _buffers = null;
         NativeMemory.Free(buffers);
+        _count = 0;
     }
 
     private void Reserve(int count)
     {
-        if (_handles.Length < count)
-        {
-            _handles = new MemoryHandle[count];
-            FreeNativeMemory();
-            _buffers = (QUIC_BUFFER*)NativeMemory.Alloc((nuint)count, (nuint)sizeof(QUIC_BUFFER));
-        }
-
+        _buffers = (QUIC_BUFFER*)NativeMemory.AllocZeroed((nuint)count, (nuint)sizeof(QUIC_BUFFER));
```

Comment:
Before this change, `Reserve` allocated only if  `_buffers` was too small to hold all `count` buffers. Now it allocates every time. Is it necessary? Is the previous `_buffers` array deallocated?

Reply:
This is just wrong, it also needs to free in case it "reallocates". I'll fix it, thanks for catching this.